File: БР1.2/Алексеев Андрей/labs/lab1/src/main/kotlin/ru/itmo/pxdkxvan/lab1/company/entity/EmployerProfileEntity.kt
package ru.itmo.pxdkxvan.lab1.company.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import org.hibernate.annotations.UuidGenerator
import ru.itmo.pxdkxvan.lab1.user.entity.UserAccountEntity
import java.time.OffsetDateTime
import java.util.UUID

@Entity
@Table(name = "employer_profile")
class EmployerProfileEntity {
    @Id
    @Column(name = "id", nullable = false, updatable = false)
    @UuidGenerator(style = UuidGenerator.Style.VERSION_7)
    var id: UUID? = null

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    var createdAt: OffsetDateTime? = null

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    var updatedAt: OffsetDateTime? = null

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    lateinit var user: UserAccountEntity

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    lateinit var company: CompanyEntity

    @Column(name = "position", nullable = false, length = 255)
    lateinit var position: String
}

File: БР1.2/Васильев Артём/labs/lab2/src/models/employer-profile.entity.ts
import {
    Column,
    Entity,
    Index,
    JoinColumn,
    ManyToOne,
    OneToMany,
} from 'typeorm';
import { Expose } from 'class-transformer';

import { AuditedEntity } from '../common/audited.entity';
import { User } from './user.entity';
import { Company } from './company.entity';
import { Vacancy } from './vacancy.entity';

@Entity({ name: 'employer_profiles' })
@Index(['userId', 'companyId'], { unique: true })
export class EmployerProfile extends AuditedEntity {
    @Expose({ name: 'user_id' })
    @Column({ name: 'user_id', type: 'uuid' })
    userId: string;

    @Expose({ name: 'company_id' })
    @Column({ name: 'company_id', type: 'uuid' })
    companyId: string;

    @Expose()
    @Column({ type: 'varchar', length: 255 })
    position: string;

    @ManyToOne(() => User, (user) => user.employerProfiles, {
        onDelete: 'CASCADE',
    })
    @JoinColumn({ name: 'user_id' })
    user: User;

    @ManyToOne(() => Company, (company) => company.employerProfiles, {
        onDelete: 'CASCADE',
    })
    @JoinColumn({ name: 'company_id' })
    company: Company;

    @OneToMany(() => Vacancy, (vacancy) => vacancy.employerProfile)
    vacancies: Vacancy[];
}
